### PR TITLE
feat: made BrowserWindow::SetBackgroundColor public

### LINKED ISSUE #13464: Add `BrowserWindow#setBackgroundColor()` method
**Is your feature request related to a problem? Please describe.**

It's possible to set the `backgroundColor` in the `BrowserWindow` constructor, but it's not possible to change it later. I need a way to dynamically change the background color when the user toggles system dark mode. I cannot just set it in CSS as my app uses `window.history.reload()`, so the original window background color becomes visible then.

**Describe the solution you'd like**

I like to see a `BrowserWindow#setBackgroundColor()` method.

The [`BrowserView`](https://github.com/electron/electron/blob/2681e769a66bad94718424be5a14492778ce674d/docs/api/browser-view.md#viewsetbackgroundcolorcolor-experimental) API already have this method.

### PR DESCRIPTION
**Pull request by darkcoderrises**
feat: made BrowserWindow::SetBackgroundColor public and implemented BrowserWindow::GetBackgroundColor
Fixes #13464

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] commit messages or PR title follow semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: atom/browser/api/atom_api_browser_window.cc [hunk 1]
```diff
@@ -293,12 +293,17 @@ void BrowserWindow::Blur() {
...
 
 void BrowserWindow::SetBackgroundColor(const std::string& color_name) {
+  background_color = color_name;
   TopLevelWindow::SetBackgroundColor(color_name);
   auto* view = web_contents()->GetRenderWidgetHostView();
   if (view)
     view->SetBackgroundColor(ParseHexColor(color_name));
 }
 
+std::string& BrowserWindow::GetBackgroundColor() {
```
Comment (MarshallOfSound) [inline atom/browser/api/atom_api_browser_window.cc]: This should return the value not the reference

File: atom/browser/api/atom_api_browser_window.h [hunk 2]
```diff
@@ -103,6 +105,8 @@ class BrowserWindow : public TopLevelWindow,
   // it should be cancelled when we can prove that the window is responsive.
   base::CancelableClosure window_unresponsive_closure_;
 
+  std::string& background_color;
```
Comment (MarshallOfSound) [inline atom/browser/api/atom_api_browser_window.h]: This should store the value not the reference

**Comment by MarshallOfSound** [comment]
Looking good @darkcoderrises 

In order to expose the getter to JS land though you will need to add a method to `atom_api_top_level_window.cc` for `GetBackgroundColor` and expose it in a similar way to the already exposed `SetBackgroundColor`.

See: https://github.com/ForkedRepos/electron/blob/e2f8501441053f4073babc1cf5405c121ad95ae6/atom/browser/api/atom_api_top_level_window.cc#L991

**Comment by darkcoderrises** [comment]
@MarshallOfSound Does this look okay?

**Comment by darkcoderrises** [comment]
Do I need to do anything more @MarshallOfSound

**Comment by codebytere** [comment]
hi @darkcoderrises could you potentially add a test for this and fill out the checklist? Once that's done we can get this across the finish line 🏁

File: atom/browser/api/atom_api_browser_window.h [hunk 3]
```diff
@@ -30,6 +30,9 @@ class BrowserWindow : public TopLevelWindow,
   static v8::Local<v8::Value> From(v8::Isolate* isolate,
                                    NativeWindow* native_window);
 
+  void SetBackgroundColor(const std::string& color_name) override;
+  std::string GetBackgroundColor();
```
Comment (miniak) [inline atom/browser/api/atom_api_browser_window.h]: this method should be `const`

File: atom/browser/api/atom_api_browser_window.h [hunk 4]
```diff
@@ -103,6 +105,8 @@ class BrowserWindow : public TopLevelWindow,
   // it should be cancelled when we can prove that the window is responsive.
   base::CancelableClosure window_unresponsive_closure_;
 
+  std::string background_color;
```
Comment (miniak) [inline atom/browser/api/atom_api_browser_window.h]: what about the default value?

Comment (darkcoderrises) [reply atom/browser/api/atom_api_browser_window.h]: What should the default value be?

Comment (ckerr) [inline atom/browser/api/atom_api_browser_window.h]: We should probably remove this variable (see above discussion on the default value) but, if this approach is what goes in, [please append a `_`](https://google.github.io/styleguide/cppguide.html#Variable_Names) to the end of the variable name.

File: atom/browser/api/atom_api_browser_window.cc [hunk 5]
```diff
@@ -293,12 +293,17 @@ void BrowserWindow::Blur() {
...
 
 void BrowserWindow::SetBackgroundColor(const std::string& color_name) {
+  background_color = color_name;
   TopLevelWindow::SetBackgroundColor(color_name);
   auto* view = web_contents()->GetRenderWidgetHostView();
   if (view)
     view->SetBackgroundColor(ParseHexColor(color_name));
 }
 
+std::string BrowserWindow::GetBackgroundColor() {
```
Comment (ckerr) [inline atom/browser/api/atom_api_browser_window.cc]: this method should be `const`

File: atom/browser/api/atom_api_browser_window.h (hunk 4 above)
Comment (ckerr) [reply atom/browser/api/atom_api_browser_window.h]: We're probably better off adding a `GetBackgroundColor()` method to `TopLevelWindow` and querying that here -- that way we wouldn't have to wonder what the default value should be.

**Comment by codebytere** [comment]
hi @darkcoderrises, please address the above comments in order to keep this pull request open.

**Comment by codebytere** [comment]
@darkcoderrises happy to reopen at some future point should you want to come back to this.

**Pull request closed without merging**